javaslang/src/main/java/javaslang/collection/Comparators.java

```diff
@@ -26,57 +26,40 @@ private Comparators() {
      * @return The natural Comparator of type U
      */
     @SuppressWarnings("unchecked")
-    static <U> SerializableComparator<U> naturalComparator() {
-        return (o1, o2) -> ((Comparable<U>) o1).compareTo(o2);
+    static <U> Comparator<U> naturalComparator() {
+        return NaturalComparator.instance();
     }
 
-    /**
-     * Needed for serialization of sortable collections which internally need a comparator.
-     * <p>
-     * In general the comparator may be
-     * <ul>
-     * <li>a concrete class</li>
-     * <li>a lambda</li>
-     * <li>a method reference</li>
-     * </ul>
-     *
-     * @param <T> the type of objects that may be compared by this comparator
-     */
-    @FunctionalInterface
-    interface SerializableComparator<T> extends Comparator<T>, Serializable {
-        long serialVersionUID = 1L;
+}
 
-        @SuppressWarnings("TrivialMethodReference")
-        static <T> SerializableComparator<T> of(Comparator<T> comparator) {
-            if (comparator instanceof SerializableComparator) {
-                return ((SerializableComparator<T>) comparator);
-            } else {
-                return comparator::compare;
-            }
-        }
```

Comment:
The result wasn't Serializable if the given comparator was not serializable. The method reference used pulled the argument into the _closure_.